typescript/codepipeline-using-cross-account-repos/index.ts

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+import 'source-map-support/register';
+import cdk = require('@aws-cdk/core');
+import codecommit= require('@aws-cdk/aws-codecommit');
+import codepipeline = require('@aws-cdk/aws-codepipeline');
+import codepipeline_actions = require('@aws-cdk/aws-codepipeline-actions');
+import iam = require('@aws-cdk/aws-iam');
+import s3 = require('@aws-cdk/aws-s3');
+import kms = require('@aws-cdk/aws-kms');
+ 
+const app = new cdk.App();
+const devEnv = {region: 'us-west-2', account: '123123123123'} 
+const toolEnv = {region: 'us-west-2', account: '776362166164'}
+ 
+// Source Stack
+const srcStack = new cdk.Stack(app, 'srcStack', { env: devEnv });
+ 
+const srcRepo = codecommit.Repository.fromRepositoryArn(srcStack, 'srcRepo', 'arn:aws:codecommit:us-west-2:363575151959:omnisearch')
+ 
+const srcRepoRole = new iam.Role(srcStack, 'CrossAccountRole', {
+  roleName: 'CrossAccountRoleName', 
+  assumedBy: new iam.AccountPrincipal(toolEnv['account']),
+});
+ 
+var srcCrossPolicy = new iam.PolicyStatement();
+srcCrossPolicy.addAllResources();
+srcCrossPolicy.addActions('codecommit:*');
+srcRepoRole.addToPolicy(srcCrossPolicy);
+ 
+//  Pipeline Stack
+const pipeStack = new cdk.Stack(app, 'pipeStack', { env: toolEnv });
+const sourceOutput = new codepipeline.Artifact();
+var pipeline = new codepipeline.Pipeline(pipeStack, 'Pipeline', {
+  artifactBucket: new s3.Bucket(pipeStack, 'Bucket', {
+    bucketName: 'castaren-test-artifact-bucket',
```

Comment:
Why are we naming a bucket? If we are marking this example as Stable, we shouldn’t be using named buckets because it’ll only deploy once before breaking.